x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/categorization/CategorizationBytesRefHash.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs.categorization;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.logging.LoggerMessageFormat;
+import org.elasticsearch.common.util.BytesRefHash;
+import org.elasticsearch.search.aggregations.AggregationExecutionException;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+class CategorizationBytesRefHash implements Closeable {
+
+    static final BytesRef WILD_CARD_REF = new BytesRef("*");
+    static final int WILD_CARD_ID = -1;
```

Comment:
These two are also probably worth javadoc. It's a bit sneaky too because `BytesRef` is a reference to an immutable underlying array. We just promise with all our hearts not to mutate it.